### PR TITLE
Make linting stricter and fix linting

### PR DESCRIPTION
**Pull request by ferferga**
This PR adds dependabot and extra linting settings to the ones added to #99. They are really similar to the ones enforced in Vue and Web

#99 must go first, as it adds extra packages needed for these settings to work, this PR just adds extra settings that are not related to the TS migration.

### CODE REVIEW

File: src/components/jellyfinActions.ts [hunk 1]
```diff
@@ -242,7 +242,7 @@ function setRandomUserBackdrop(): Promise<void> {
             // not everyone will want to see adult backdrops rotating on their TV.
             MaxOfficialRating: 'PG-13'
         }
-    }).then(function (result: BaseItemDtoQueryResult) {
+    }).then((result: BaseItemDtoQueryResult) => {
```
Comment (hawken93) [inline src/components/jellyfinActions.ts:245]: Can you fix the new lint errors as well? Ideally PRs that fail checks should not be merged, hence I at least expect the new linting checks to also be corrected for by this PR. If you have time, the other linting checks as well.

In this instance this function should be rewritten to await syntax.

File: src/components/jellyfinActions.ts [hunk 2]
```diff
@@ -381,11 +381,21 @@ export function play($scope: GlobalScope): void {
...
+/**
+ * @param item
+ * @param maxBitrate
+ * @param deviceProfile
+ * @param startPosition
+ * @param mediaSourceId
+ * @param audioStreamIndex
+ * @param subtitleStreamIndex
+ * @param liveStreamId
+ */
```
Comment (hawken93) [inline src/components/jellyfinActions.ts:398]: Don't check this in without completing the JSDoc please, it's not good to add this just to squelsh the lint error/warning, also the one just below.

File: src/components/jellyfinActions.ts [hunk 3]
```diff
@@ -506,12 +526,12 @@ export function getDownloadSpeed(byteSize: number): Promise<number> {
  */
 export function detectBitrate(): Promise<number> {
```
Comment (hawken93) [inline src/components/jellyfinActions.ts:527]: For this function my PR changes to await syntax. So no changes necessary just in this instance.

File: src/components/maincontroller.ts [hunk 4]
```diff
@@ -48,6 +48,9 @@ let broadcastToServer = new Date();
 
 let hasReportedCapabilities = false;
 
+/**
+ *
+ */
```
Comment (hawken93) [inline src/components/maincontroller.ts:53]: jsdoc needs to be completed, same for all the following ones.

**Review by hawken93 — CHANGES_REQUESTED** [review]
Looks good, except my annoyance that I'd like the changes to go with the linting so that we're not buried deep in  lint warnings.

Even before this there are some 260 linter warnings/errors and this will increase the errors even more. It will decrease the warnings some due to incomplete JSDoc but I'd rather have linter warnings than empty comment blocks. I think it's a bad thing that eslint even autofixes to empty comment blocks and params without types or descriptions, these should all be warnings too, but they are not.

In such a situation, new code may introduce more linter remarks, not less, because the linter loses its functionality as a mechanism to keep the code clean.


Also, if you know any checks to make sure public class members in typescript also have documentation, then I think this should be added as well.

I understand that you may not be able to make good jsdoc comments, but in this regard we can prioritize not having any errors, and I'll fix jsdoc that requires knowing the code over time.

**Comment by ferferga** [comment]
@hawken93 will do once we finish all the huge refactoring you did. Didn't want to mess with that until everything was in place 👍🏻.

**Comment by ferferga** [comment]
Using all the identation settings from Vue and fixed all the linting errors (warnings from JSDoc are still there)

Most of the remaining earnings can be fixed as soon as we have proper typings with axios client.

Marking as ready to review.

File: .prettierrc [hunk 5]
```diff
@@ -1,6 +1,5 @@
 {
-    "singleQuote": true,
-    "trailingComma": "none",
-    "semi": true,
-    "tabWidth": 4
+  "semi": true,
+  "singleQuote": true,
+  "trailingComma": "none"
```
Comment (Artiume) [inline .prettierrc:4]: did we mean to lose tabWidth?

File: src/components/codecSupportHelper.ts [hunk 6]
```diff
@@ -216,9 +221,9 @@ export function getSupportedMP4AudioCodecs(): Array<string> {
  * @returns Supported HLS video codecs.
  */
 export function getSupportedHLSVideoCodecs(): Array<string> {
-    // Currently the server does not support fmp4 which is required
-    // by the HLS spec for streaming H.265 video.
-    return ['h264'];
+  // Currently the server does not support fmp4 which is required
+  // by the HLS spec for streaming H.265 video.
```
Comment (Artiume) [inline src/components/codecSupportHelper.ts:225]: is this still true?

File: .prettierrc (hunk 5 above)
Comment (ferferga) [reply .prettierrc:4]: The defaults of Prettier are 2 (so don't need to be explicit about this parameter)

2 tabs is more consistent to what we use in other projects.

It's also enforced in .editorconfig so IDEs are aware of this

File: src/components/codecSupportHelper.ts (hunk 6 above)
Comment (ferferga) [reply src/components/codecSupportHelper.ts:225]: Don't know as I did the tests just to check that nothing broke while changing between promises to async/await syntax (which was all that was needed to fix the linting and modernize the code according to our other projects)

This is outside the scope of this PR and should be determined later imo.

File: .prettierrc (hunk 5 above)
Comment (hawken93) [reply .prettierrc:4]: As we chatted about on matrix, I would prefer to keep 4, although I would be able to accept 2 space indent.

File: .eslintrc.js [hunk 7]
```diff
@@ -1,26 +1,90 @@
 module.exports = {
     root: true,
-    plugins: ['@typescript-eslint', 'promise', 'import', 'jsdoc'],
     env: {
         node: true,
-        es6: true,
-        browser: true
+        browser: true,
+        es6: true
```
Comment (hawken93) [inline .eslintrc.js:6]: There are some reorderings that are still not alphabetical, maybe sort by alphabet?

File: src/components/documentManager.ts [hunk 8]
```diff
@@ -129,65 +132,65 @@ export abstract class DocumentManager {
...
-                } else if (item?.UserData?.UnplayedItemCount) {
-                    this.setPlayedIndicator(item?.UserData?.UnplayedItemCount);
-                } else {
-                    this.setPlayedIndicator(false);
-                }
+        const urls = [
+            await this.getWaitingBackdropUrl(item),
+            await this.getPrimaryImageUrl(item),
+            await this.getLogoUrl(item)
+        ];
```
Comment (hawken93) [inline src/components/documentManager.ts:151]: I'd like to keep the same behavior here, I did have a shot at it in https://github.com/jellyfin/jellyfin-chromecast/pull/190/files#diff-f7d8d074ad9f2f33a0ea4c9774f0aca90b80d72e4e8dba4f79dcdfc9283228d6R144-R183

Basically we switch to async/await but keep Promise.all() around.

Comment (hawken93) [reply src/components/documentManager.ts:151]: It's network requests so doing 3 in parallell is much better than 3 in serial..

File: src/components/jellyfinActions.ts [hunk 9]
```diff
@@ -292,6 +306,16 @@ export function getPlaybackInfo(
...
+/**
+ * @param item
+ * @param playSessionId
+ * @param maxBitrate
+ * @param deviceProfile
+ * @param startPosition
+ * @param mediaSource
+ * @param audioStreamIndex
+ * @param subtitleStreamIndex
+ */
```
Comment (hawken93) [inline src/components/jellyfinActions.ts:318]: maybe remove this too, then me or someone who actually documents the function can take away the warning

File: src/components/jellyfinActions.ts (hunk 2 above)
Comment (hawken93) [reply src/components/jellyfinActions.ts:262]: There should now be a lint rule to warn on empty blocks as well, so this should still produce a warning

File: src/components/maincontroller.ts (hunk 4 above)
Comment (hawken93) [reply src/components/maincontroller.ts:51]: There should now be a lint rule to warn on empty blocks as well, so this should still produce a warning

File: .prettierrc (hunk 5 above)
Comment (hawken93) [reply .prettierrc]: It's now 4 spaces, ty :)

File: .editorconfig [hunk 10]
```diff
@@ -3,7 +3,7 @@ root = true
 [*]
 indent_style = space
 indent_size = 4
+end_of_line = lf
 charset = utf-8
 trim_trailing_whitespace = true
 insert_final_newline = true
```
Comment (hawken93) [inline .editorconfig:9]: maybe sort this block as well?

File: src/components/documentManager.ts (hunk 8 above)
Comment (hawken93) [reply src/components/documentManager.ts:151]: Yey :)

File: src/components/codecSupportHelper.ts (hunk 6 above)
Comment (hawken93) [reply src/components/codecSupportHelper.ts]: As far as I can see, this file does not have any changes relating to this code anymore

**Review by hawken93 — APPROVED** [review]

**Comment by hawken93** [comment]
:+1:

**Pull request merged**